src/verify.rs

```diff
@@ -197,6 +202,64 @@ pub fn verify_server_cert(roots: &RootCertStore,
     .map_err(|err| TLSError::WebPKIError(err))
 }
 
+pub fn parallel_verify_server_cert(roots: &RootCertStore,
+                                   presented_certs: &Vec<ASN1Cert>,
+                                   dns_name: &str) -> Result<(), TLSError> {
+  crossbeam::scope(|scope| {
+    scope.spawn(move || {
+      parallel_verify_common_cert(roots, presented_certs);
+    });
+    let r = scope.spawn(move || {
+      let cert_der = untrusted::Input::from(&presented_certs[0].0);
+      let cert = try!(
+        webpki::EndEntityCert::from(cert_der)
+          .map_err(|err| TLSError::WebPKIError(err))
+      );
+      cert.verify_is_valid_for_dns_name(untrusted::Input::from(dns_name.as_bytes()))
+      .map_err(|err| TLSError::WebPKIError(err))
+    });
+    r.join()
+  })
+  
+}
+
+fn parallel_verify_common_cert<'a>(roots: &RootCertStore,
```

Comment:
there's a minute change--I'm using par_iter instead.

When I was looking at performance, parallelizing wasn't improving the code at all, so parallel_verify_common_cert uses rayon to convert from rustls to webpki structures. After that, performance improved.  I didn't necessarily want to make permanent changes to the serial code (but I can).